### PR TITLE
feat: role subscription metadata and channel flags

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

https://github.com/discord/discord-api-docs/pull/5828

Part 2 of (and blocked by) #904.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Victorsitou** [comment]
Why is this a separate PR and not included in #904? 🤔

File: docs/api.rst [hunk 1]
```diff
@@ -5763,6 +5769,14 @@ InteractionReference
 .. autoclass:: InteractionReference
     :members:
 
+RoleSubscriptionData
```
Comment (Victorsitou) [inline docs/api.rst:5772]: Isn't this a Discord Model? This is currently under data classes

**Comment by shiftinv** [comment]
> Why is this a separate PR and not included in #904?

I didn't expect the api docs PR to be merged this quickly :p (there are a couple PRs that have been waiting for weeks, like #892)

File: docs/api.rst (hunk 1 above)
Comment (shiftinv) [reply docs/api.rst]: Yup, I mistakenly just put it right next to `InteractionReference`, which turned out to also be in the wrong section.
moved them in https://github.com/DisnakeDev/disnake/pull/913/commits/b150cd14973934926f8f5da7f67c5386a7e33351

**Comment by shiftinv** [comment]
Rebased on master now that #904 is merged.

File: disnake/message.py [hunk 2]
```diff
@@ -1250,6 +1291,17 @@ def thread(self) -> Optional[Thread]:
 
         return self.guild.get_thread(self.id)
 
+    @property
```
Comment (onerandomusername) [inline disnake/message.py:1294]: use cached_property? or is RoleSubscriptionData mutable by the user

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/message.py (hunk 2 above)
Comment (shiftinv) [reply disnake/message.py:1294]: I shouldn't be mutated, but is it worth storing this in memory for the lifetime of the Message object?

Comment (onerandomusername) [reply disnake/message.py:1294]: ... fair, messages can be cached for a long time.

**Comment by onerandomusername** [comment]
@shiftinv would you please resolve conflicts?

**Review by onerandomusername — APPROVED** [review]
never approved, conflicts were resolved

**Pull request merged**